File: src/Frontend/src/models/Turn.ts
import type { AgentRole } from "./AgentRole.ts";
import type { TurnStatus } from "./TurnStatus.ts";

export interface Turn {
  code: string;
  role: AgentRole;
  createdAt: string;
  expiresAt: string | null;
  status: TurnStatus;
}

File: src/Backend/Models/Turn.cs
namespace QueuingSystem.Backend.Models;

public class Turn
{
    public Guid Id { get; init; }
    public required string Code { get; init; }
    public required string Role { get; init; }
    public DateTime CreatedAt { get; init; }
    public DateTime? CompletedAt { get; set; }
    public TurnStatus Status { get; set; }
}
